rotkehlchen/api/rest.py

```diff
@@ -4719,25 +4719,30 @@ def export_history_events(
 
         serialized_history_events = []
         headers: dict[str, None] = {}
+        prices_from_db = GlobalDBHandler.get_historical_prices(
+            query_data=(query_data := [(event.asset, currency, ts_ms_to_sec(event.timestamp)) for event in history_events]),  # type: ignore[arg-type]  # currency is a subclass of Asset  # noqa: E501
+            max_seconds_distance=HOUR_IN_SECONDS,
+        )
+        missing_prices: list[tuple[Asset, Timestamp]] = []
+        cached_db_prices: defaultdict[Asset, defaultdict[Timestamp, FVal]] = defaultdict(lambda: defaultdict(lambda: ZERO))  # noqa: E501
+        for idx, data in enumerate(query_data):
+            if (db_price := prices_from_db[idx]) is None or (data[0], data[2]) not in missing_prices:  # noqa: E501
+                missing_prices.append((data[0], data[2]))
+            else:
+                cached_db_prices[db_price.from_asset][db_price.timestamp] = db_price.price
+
+        queried_prices = PriceHistorian.query_multiple_prices(
+            assets_timestamp=missing_prices,
+            target_asset=currency,
+            msg_aggregator=self.rotkehlchen.msg_aggregator,
+        )
         for event in history_events:
-            try:  # ask oracles for the price in the given timestamp and currency
-                price = PriceHistorian.query_historical_price(
-                    from_asset=event.asset,
-                    to_asset=currency,
-                    timestamp=ts_ms_to_sec(event.timestamp),
-                )
-            except (PriceQueryUnsupportedAsset, RemoteError):
-                fiat_value = ZERO
-            except NoPriceForGivenTimestamp as e:
-                # In the case of NoPriceForGivenTimestamp when we got rate limited
-                if e.rate_limited is True:
-                    return wrap_in_fail_result(
-                        message='Price query got rate limited for all the oracles. Try again later',  # noqa: E501
-                        status_code=HTTPStatus.BAD_GATEWAY,
-                    )
-                fiat_value = ZERO
+            if (cached_price := cached_db_prices[event.asset][ts_ms_to_sec(event.timestamp)]) != ZERO:  # noqa: E501
+                fiat_value = cached_price * event.amount
+            elif (queried_price := queried_prices[event.asset][ts_ms_to_sec(event.timestamp)]) != ZERO:  # noqa: E501
+                fiat_value = queried_price * event.amount
             else:
-                fiat_value = event.amount * price
+                fiat_value = ZERO
 
             serialized_event = event.serialize_for_csv(
                 fiat_value=fiat_value,
```

Comment:
```suggestion
            serialized_event = event.serialize_for_csv(
                fiat_value=event.amount * (
                    cached_db_prices[event.asset][ts_ms_to_sec(event.timestamp)] or
                    queried_prices[event.asset][ts_ms_to_sec(event.timestamp)]
                )
```
I think doing something like this should work, and then the ifs above can be removed.